### ISSUE TITLE
Container not found with WordPress

### ISSUE THREAD

**Issue by na86** [luruke/barba.js#25]
Hi guys, 
thanks for the amazing work done here. 
I'm encountering some issue with barba.js and wordpress
i'm buliding a simple and lightweight website with a custom theme (starting form scratch).

i've tried to follow your instructions but the console is returning me the following 
`
> Uncaught (in promise) Error: Barba.js: no container found(…)  - barba.js?ver=4.5.2:1447

Where can i find a more complete guide? 
Thanks a lot for your beautiful work

Nan
`

**Comment by luruke** [comment]
Hello @na86!
Thank you for trying Barba.js

I've used barba with WordPress in few project and there are no problems with it.

From the error I can assume you are trying to load a page that doesn't contain `.barba-container`. (Or another class if you changed it)

Are you sure yours template are using the correct Dom structure?

Are you logged as admin in WordPress and clicking (or hovering if the prefetch is active) on the admin bar?

Usually I disable barba.js when the user is logged as administrator in order to avoid this problems.

**Comment by luruke** [comment]
I changed the title to be more appropriate

**Comment by na86** [comment]
Thank you Luigi, 
it works!!!

(i'm used to develop wordpress site in the ordinary way, this is a kind of adventure for me.)

p.s.: sorry for the title, thankyou for replacing it.

**Comment by paulzz3000** [comment]
Hi Nan, do you have any tips for adding this to Wordpress?
Thanks

**Comment by luruke** [comment]
Hi @paulzz3000 , do you have any specific problem?

Usually for the DOM structure, in the header file I add something like:

```html
<head>...</head>
<body>
    <main id="barba-wrapper">
        <div data-namespace="<?php echo get_current_template() ?>" <?php body_class('barba-container') ?>
```

where get_template_file() is something like
```php
function get_current_template() {
  global $template;
  return basename($template, '.php');
}
```

and then in the footer I just close my barba-container and barba-wrapper.

**Comment by dblodorn** [comment]
This is awesome @luruke - and works great with Wordpress (almost to easy). I run into an issue though with using the wordpress menu outside of the barba-wrapper, the active state on links does not get updated accordingly. I'm using a css navwalker function in functions.php to update the active state on each link after it gets clicked, but this function only gets called once if it is outside of the barba wrapper, any ideas regarding this, is it safe to call a php function with javascript? Or would that even be the right direction to go in? Thanks!

**Comment by luruke** [comment]
Hi @dblodorn .

You can:
- Get all the HTML of the new page, and detect which is the active link, then via js update accordingly: https://github.com/luruke/barba.js/issues/49
- Add a data parameter on the `.barba-container` with the current menu id, then via js update accordingly reading the data attribute `newPageReady` event

**Comment by KushnerMikalai** [comment]
If I want not all pages to be without reboot, how can I do this? Now, when I turn to a page without a container, does it give me this error (Uncaught (in promise) Error: Barba.js: no container found.....) ???

**Comment by luruke** [comment]
@KushnerMikalai ,
ideally all the pages of your website should use `barba-container`.

If you can't, you should add a check with `initStateChange` and force the page reload to the newer URL that do not have barba-container

**Comment by ganesanshivam** [comment]
Please provide a code example...

**Comment by kajheijmans** [comment]
I know this is an old thread; just wanted to let you know (for the sake of others maybe) I ran into the same problem and fixed it by using the enqueue script for the footer instead. 
The issue is has a simple cause: BarbaJS doesn't know the container exists because at that point, it doesn't (before DOM load).

tl;dr; set the last parameter in your enqueue_script function (where barba functions are loaded) to true.

https://developer.wordpress.org/reference/functions/wp_enqueue_script/
